clippy_lints/src/casts/mod.rs

```diff
@@ -251,6 +252,36 @@ declare_clippy_lint! {
     "casting a function pointer to a numeric type not wide enough to store the address"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for casts of a function pointer to **any** numeric type.
+    ///
+    /// ### Why is this bad?
+    /// Casting a function pointer to an integer can have surprising results and can occur
+    /// accidentally if parantheses are omitted from a function call. If you aren't doing anything
+    /// low-level with function pointers then you can opt-out of casting functions to integers to
+    /// avoid mistakes. Alternatively, you can use this lint to audit all uses of function pointer
+    /// to integer casts in your code.
+    ///
+    /// ### Example
+    /// ```rust
+    /// // Bad: forgot to call `fn1` before casting to `usize`
+    /// fn fn1() -> u16 {
+    ///     1
+    /// };
+    /// let _ = fn1 as usize;
+    ///
+    /// // Good: lint ensures that you remember to call `fn2`
```

Comment:
This should be re-worded since the lint cannot assume that the coder intended to call the function.

Reply:
Done